WebNetSample/Controllers/ProductsController.cs

```diff
@@ -18,4 +22,41 @@ public async Task<IActionResult> Index()
 
         return View(productsWithDetails);
     }
+
+    public async Task<IActionResult> AllProducts()
```

Comment:
1. Let's add HTTP method attribute above to make all actions consistent:
2. Let's rename action to `GetProducts` to make its name clear and satisfy code style best practices (don't forget to update usages in views if they refer to this endpoint);
3. Your method name does not satisfy its purpose exactly: you have name `AllProducts`, but you are performing pagination search. This is too confusing, especially for other devs. Let's avoid such mistypes. I have shared idea how to improve it